docs/core/development/policy/coding-standard.rst

```diff
@@ -625,6 +625,126 @@ Any attribute, method argument, or method name that corresponds *directly* to a
 All SQL keywords should be in upper case.
 
 
+Emitting logs
+-------------
+
+All new code emitting log messages should use the new logging system.
+
+This section is dedicated to using the new logging system in Twisted core code.
+For general information about using the logging system please check the dedicated documentation about :doc:`how to log with twisted.logger </core/howto/logger>` and the :api:`twisted.logger.Logger <Logger>` API docs.
+
+General usage rules:
+
+* Don't use explicit namespaces.
+  The logger will automatically fill the namespace.
+* Use the global log publisher.
+  Twisted code may call into non-Twisted code which makes stdlib/legacy log messages, which then get redirected to the global logger.
+* When possible instantiate the Logger as ``self._log`` instance member.
+* Emitted logs are not part of the API/contract so you don't have to document it as part of the API.
+* For security features you might documented the emitted logs and then they will became part of the contract.
+
+Example of usage:
+
+.. code-block:: python
+
+    from twisted.logger import Logger
+
+    class SomeStuff(object):
+        """
+        Class using the new logging system as an instance member
+        """
+
+        _log = Logger()
+
+        def getUser(self, name):
+            try:
+                return self._otherStuff()
+            except Exception as error:
+                self._log.info(
```

Comment:
This is an error: any `except Exception` should be logged with `self._log.failure()`; by doing this we've discarded the traceback and prevented the user from being able to debug with the logs.